src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -196,13 +198,29 @@ See the LICENSE file in the project root for more information.
     </ItemGroup>
 
     <MakeDir Directories="$(NativeIntermediateOutputPath)" />
-    <WriteLinesToFile File="$(NativeIntermediateOutputPath)%(ManagedBinary.Filename).ilc.rsp" Lines="@(IlcArg)" Overwrite="true" />
+
+    <ItemGroup Condition="'$(NativeCodeGen)' != 'readytorun'">
+      <IlcCmd Include="$(ManagedBinaryFilename)">
+        <ResponseFileName>$(ManagedBinaryFilename).ilc.rsp</ResponseFileName>
+        <CommandLine>@(IlcCompileInput);-o:$(NativeIntermediateOutputPath)$(ManagedBinaryFilename)$(IlcOutputFileExt);@(IlcArg)</CommandLine>
+      </IlcCmd>
+    </ItemGroup>
+
+    <ItemGroup Condition="'$(NativeCodeGen)' == 'readytorun'">
+      <IlcCompileInput2 Include="@(IlcCompileInput)" />
+      <IlcCmd Include="@(IlcCompileInput)">
+        <ResponseFileName>%(IlcCompileInput.FileName).ilc.rsp</ResponseFileName>
+        <CommandLine>%(IlcCompileInput.Identity);@(IlcCompileInput2 -> '-r:%(Identity);');-o:$(NativeIntermediateOutputPath)%(IlcCompileInput.Filename).ni%(IlcCompileInput.Extension);@(IlcArg)</CommandLine>
```

Comment:
I'm a bit confused reading this - What does duplicating `IlcCompileInput` into `IlcCompileInput2` and mixing the two buy us?

Reply:
Oh, it looks like maybe we're permuting each input and having all the other inputs as references so we compile N times? I think a comment here would be a good idea as this would be easy to break in the future inadvertently.

Reply:
Exactly - when we're within a loop over %(IlcCompileInput.Identity), we cannot expand the item group "again" so that's why I'm copying it. I'll add an explanatory comment, that's definitely a good idea.